### PR TITLE
Feature/add teacher get courses

### PR DESCRIPTION
**Pull request by zamiramanasova**
add teachers courses

### CODE REVIEW

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java [hunk 1]
```diff
@@ -52,4 +57,12 @@ public ResponseEntity<String> delete(@PathVariable Long id) {
         teacherService.delete(id);
         return new ResponseEntity<>("Teacher deleted successfully.", HttpStatus.OK);
     }
+
+    @JsonIgnore
+    @GetMapping("/courses/{username}")
```
Comment (peaksoft-school) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java:62]: Should be ("courses") only

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java [hunk 2]
```diff
@@ -52,4 +57,12 @@ public ResponseEntity<String> delete(@PathVariable Long id) {
...
+    @PreAuthorize("hasAnyAuthority('ROLE_INSTRUCTOR','ROLE_ADMIN')")
```
Comment (peaksoft-school) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java:63]: Should be 'ROLE_INSTRUCTOR' only

**Review by peaksoft-school — COMMENTED** [review]
Comments

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/CourseController.java [hunk 3]
```diff
@@ -12,42 +13,49 @@
...
+@Tag(name = "Course controller", description = "ADMIN create, update, and delete")
 public class CourseController {
 
     private final CourseService courseService;
 
     @PostMapping
     @PreAuthorize("hasAnyAuthority('ROLE_ADMIN')")
+    @Operation(summary = "method create", description = "admin can create course")
     public ResponseEntity<CourseResponce> create(@RequestBody @Valid CourseRequest request){
         return new ResponseEntity<>(courseService.save(request), HttpStatus.CREATED);
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/CourseController.java:30]: Should return HttpStatus Ok

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/CourseController.java [hunk 4]
```diff
@@ -12,42 +13,49 @@
...
+@CrossOrigin(origins = "*", allowedHeaders = "*",maxAge = 3600)
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("api/courses")
+@Tag(name = "Course controller", description = "ADMIN create, update, and delete")
 public class CourseController {
 
     private final CourseService courseService;
 
     @PostMapping
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/CourseController.java:26]: Write top of method

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/CourseController.java [hunk 5]
```diff
@@ -12,42 +13,49 @@
...
     @PreAuthorize("hasAnyAuthority('ROLE_ADMIN')")
+    @Operation(summary = "method get by id", description = "admin can get by id course")
     public ResponseEntity<CourseResponce> getById(@PathVariable Long id) {
         return ResponseEntity.ok(courseService.getById(id));
     }
 
     @DeleteMapping("{id}")
     @PreAuthorize("hasAnyAuthority('ROLE_ADMIN')")
+    @Operation(summary = "method delete", description = "admin can delete course")
     public ResponseEntity<String> delete(@PathVariable Long id) {
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/CourseController.java:51]: Should return CourseResponse

File: src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java [hunk 6]
```diff
@@ -23,33 +27,41 @@ public class TeacherController {
...
+    @GetMapping("{id}")
     public ResponseEntity<TeacherResponse> getById(@PathVariable Long id) {
         return ResponseEntity.ok(teacherService.getById(id));
     }
 
-    @DeleteMapping("{id}")
     @PreAuthorize("hasAnyAuthority('ROLE_ADMIN')")
     @Operation(summary = "method delete", description = "admin can delete teacher")
+    @DeleteMapping("{id}")
     public ResponseEntity<String> delete(@PathVariable Long id) {
```
Comment (ChyngyzIshen) [inline src/main/java/kg/peaksoft/peaksoftlmsm1/api/TeacherController.java:56]: Should return TeacherResponse

**Review by ChyngyzIshen — COMMENTED** [review]
comments

**Pull request merged**